edgehog-device-runtime-docker/src/image.rs

```diff
@@ -0,0 +1,553 @@
+            None => (None, rest),
+        };
+
+        Ok(Image { name, tag, repo }.into())
+    }
+}
+
+impl TryFrom<ImageSummary> for Image<String> {
+    type Error = ConversionError;
+
+    fn try_from(value: ImageSummary) -> Result<Self, Self::Error> {
+        let tag = value
+            .repo_tags
+            .first()
+            .ok_or(ConversionError::UntaggedImage)?;
+
+        tag.parse()
+    }
+}
+
+impl From<Image<&str>> for Image<String> {
+    fn from(value: Image<&str>) -> Self {
+        Image {
+            name: value.name.to_string(),
+            tag: value.tag.to_string(),
+            repo: value.repo.map(ToString::to_string),
+        }
+    }
+}
+
+impl<S> Image<S> {
+    /// Create a new docker image.
+    pub fn new(name: S, tag: S) -> Self {
+        Self {
+            name,
+            tag,
+            repo: None,
+        }
+    }
+
+    /// Create an image with tag and repo.
+    pub fn with_repo(name: S, tag: S, repo: S) -> Self {
+        Self {
+            name,
+            tag,
+            repo: Some(repo),
+        }
+    }
+
+    /// Get a reference to the image
+    pub fn tag(&self) -> String
+    where
+        S: Display,
+    {
+        self.to_string()
+    }
+
+    /// Pull the docker image struct.
+    ///
+    /// See the [Docker API reference](https://docs.docker.com/engine/api/v1.43/#tag/Image/operation/ImageCreate)
+    #[instrument]
+    pub async fn pull(&self, client: &Client) -> Result<(), ImageError>
+    where
+        S: Debug + Display,
+    {
+        let options = CreateImageOptions::from(self);
+
+        debug!("Creating the image {}", self);
+
+        client
+            // NOTE: Create the image by pulling it from the registry, here it's missing the root fs
+            //       option and the registry auth
+            .create_image(Some(options), None, None)
+            .try_for_each(|create_info| {
+                debug!("creating image: {:?}", create_info);
+
+                future::ready(Ok(()))
+            })
+            .await
+            .map_err(ImageError::Pull)?;
+
+        info!("Image {} created", self);
+
+        Ok(())
+    }
+
+    /// Inspect a docker image.
+    ///
+    /// See the [Docker API reference](https://docs.docker.com/engine/api/v1.43/#tag/Image/operation/ImageInspect)
+    #[instrument]
+    pub async fn inspect(&self, client: &Client) -> Result<Option<ImageInspect>, ImageError>
+    where
+        S: Debug + AsRef<str>,
+    {
+        let res = client.inspect_image(self.name.as_ref()).await;
+
+        debug!("received response {res:?}");
+
+        // Check if the image was not found
+        let inspect = match res {
+            Ok(inspect) => inspect,
+            Err(BollardError::DockerResponseServerError {
+                status_code: 404,
+                message,
+            }) => {
+                debug!("image not found: {message}");
+
+                return Ok(None);
```

Comment:
Why do you return Ok here? Is it handled in a different way ?I refer to `404` error code, seems that image was not found.

Reply:
The error 404 is returned for an image not found